src/rosdiscover/interpreter/interpreter.py

```diff
@@ -21,22 +22,23 @@ class Interpreter:
     """
     @classmethod
     @contextlib.contextmanager
-    def for_image(cls,
-                  image: str,
-                  sources: Sequence[str],
-                  *,
-                  environment: Optional[Mapping[str, str]] = None
-                  ) -> Iterator['Interpreter']:
-        """Constructs an interpreter for a given Docker image."""
+    def for_config(cls,
+                   config: Config
+                   ) -> Iterator['Interpreter']:
+        """Constructs an interpreter for a given configuration"""
         rsw = roswire.ROSWire()  # TODO don't maintain multiple instances
-        with rsw.launch(image, sources, environment=environment) as app:
+        with rsw.launch(config.image, config.sources, environment=config.environment) as app:
             yield Interpreter(app)
 
     def __init__(self, app: roswire.System) -> None:
         self._app = app
         self.params = ParameterServer()
         self.nodes: Dict[str, NodeContext] = {}
 
+    @property
+    def system(self) -> roswire.System:
```

Comment:
We should probably rename this to `AppInstance` and use `roswire.AppInstance` rather than `roswire.System` (which is deprecated).

Reply:
OK. I think I was copying it from some ROSWire code, that is probably still using the deprecated stuff. Fixed now.